### PR TITLE
Minor fix for riscv

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Address the last comment in #5782

## Impact
Minor, rename and style change

## Testing
Pass CI

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@pussuw could you take a look?

**Review by pkarashchenko — APPROVED** [review]

File: arch/risc-v/src/common/supervisor/Make.defs [hunk 1]
```diff
@@ -21,7 +21,7 @@
 # If the NuttX kernel runs in S-mode
 
 CMN_ASRCS += riscv_dispatch_syscall.S
-CMN_CSRCS += riscv_sbi.c riscv_perform_syscall.c
+CMN_CSRCS += riscv_perform_syscall.c
```
Comment (pussuw) [inline arch/risc-v/src/common/supervisor/Make.defs:24]: Oops, I'll provide the file on monday

File: arch/risc-v/src/common/riscv_percpu.c [hunk 2]
```diff
@@ -61,11 +61,11 @@ static struct riscv_percpu_s g_scratch[HART_CNT];
 
 void riscv_percpu_init(void)
 {
-  int i;
+  uintptr_t i;
```
Comment (pussuw) [inline arch/risc-v/src/common/riscv_percpu.c:64]: I think using int as iterator is quite common in the code, but using stdint types is great IMO

File: arch/risc-v/src/common/riscv_percpu.c [hunk 3]
```diff
@@ -111,8 +111,8 @@ uintptr_t riscv_percpu_get_hartid(void)
 {
   uintptr_t scratch = READ_CSR(CSR_SCRATCH);
 
-  DEBUGASSERT(scratch >= (uintptr_t) &g_scratch &&
-              scratch <= (uintptr_t) &g_scratch + sizeof(g_scratch));
+  DEBUGASSERT(scratch >= (uintptr_t) &g_percpu &&
+              scratch < (uintptr_t) &g_percpu + sizeof(g_percpu));
```
Comment (pussuw) [inline arch/risc-v/src/common/riscv_percpu.c:115]: Thanks for fixing this bug

**Review by pussuw — APPROVED** [review]
LGTM!

File: arch/risc-v/src/common/supervisor/Make.defs (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/risc-v/src/common/supervisor/Make.defs:24]: Sure, let's wait your new patch to bring riscv_sbi.c back

File: arch/risc-v/src/common/riscv_percpu.c (hunk 2 above)
Comment (xiaoxiang781216) [reply arch/risc-v/src/common/riscv_percpu.c:64]: I change i to uintptr_t because hartid is uintptr_t too.

**Pull request merged**